### ISSUE TITLE
"The TypeScript language service died" on specific JS code

### ISSUE THREAD

**Issue by ShingarevS** [microsoft/vscode#88430]
Issue Type: <b>Bug</b>

The TypeScript language service died 5 times right after it got started. The service will not be restarted.
Version: 1.41.1 (system setup)
Commit: 26076a4de974ead31f97692a0d32f90d735645c0
Date: 2019-12-18T14:58:56.166Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362

To reproduce:
1. Create a file tst.js:
```
const Router = require('restify-router').Router;
const router = new Router();
const formidable = require('formidable');

router.post('/user_profile', (req, res, next) => {
    var form = new formidable.IncomingForm();
    form.parse(req)
        .on('field', (name, field) => {
            this[name] = field;
        })
});
module.exports = router;
```
2. Open in VS Code
3. Restart VS.
TS Server crashes on initialization. If comment the row 
` this[name] = field;`
or change it to
` this.name = field;`
the issue doesn't appear.
Disabling extensions doesn't help.

TS Server log attached
[tslog.txt](https://github.com/microsoft/vscode/files/4046093/tslog.txt)

**Comment by mjbvz** [comment]
Confirmed in current release (which uses TS 3.7)

Fixed in TypeScript 3.8 (TS powers our JS intellisense as well). To workaround, update your workspace to use `typescript@next` by installing [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next)

**Comment by mjbvz** [comment]
Relevant error stack:

```
Debug Failure. False expression.Error: Debug Failure. False expression.    at declareSymbol (tsserver.js:29212:22)    at bindThisPropertyAssignment (tsserver.js:31366:25)    at bindWorker (tsserver.js:31020:29)    at bind (tsserver.js:30902:13)    at bindExpressionStatement (tsserver.js:30106:13)    at bindChildrenWorker (tsserver.js:29568:21)    at bindChildren (tsserver.js:29489:17)    at bind (tsserver.js:30913:21)    at tsserver.js:29495:94    at bindEach (tsserver.js:29511:21)    at bindEachFunctionsFirst (tsserver.js:29495:13)    at bindChildrenWorker (tsserver.js:29611:21)    at bindChildren (tsserver.js:29489:17)    at bind (tsserver.js:30913:21)    at visitNode (tsserver.js:17622:24)    at Object.forEachChild (tsserver.js:17752:21)    at bindEachChild (tsserver.js:29519:16)    at bindChildrenWorker (tsserver.js:29614:21)    at bindChildren (tsserver.js:29489:17)    at bindContainer (tsserver.js:29435:17)    at bind (tsserver.js:30916:21)    at bindEach (tsserver.js:29511:21)    at visitNodes (tsserver.js:17627:24)    at Object.forEachChild (tsserver.js:17815:21)    at bindEachChild (tsserver.js:29519:16)    at bindCallExpressionFlow (tsserver.js:30360:21)    at bindChildrenWorker (tsserver.js:29596:21)    at bindChildren (tsserver.js:29489:17)    at bind (tsserver.js:30913:21)    at bindExpressionStatement (tsserver.js:30106:13)    at bindChildrenWorker (tsserver.js:29568:21)    at bindChildren (tsserver.js:29489:17)    at bind (tsserver.js:30913:21)    at tsserver.js:29495:94    at bindEach (tsserver.js:29511:21)    at bindEachFunctionsFirst (tsserver.js:29495:13)    at bindChildrenWorker (tsserver.js:29611:21)    at bindChildren (tsserver.js:29489:17)    at bind (tsserver.js:30913:21)
```
